src/averageAges.js

```diff
@@ -37,7 +52,18 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const women = withChildren
+    ? people.filter(person => person.sex === 'f'
+      && people.find(children => children.mother === person.name))
+    : people.filter(person => person.sex === 'f');
```

Comment:
see the comment below about repeating the logic - move the `withChildren` condition inside `.filter`